topiary-core/src/atom_collection.rs

```diff
@@ -535,15 +560,38 @@ impl AtomCollection {
             self.atoms.push(Atom::Leaf {
                 content: String::from(node.utf8_text(source)?),
                 id,
-                original_position: node.start_position().into(),
+                original_column: node.start_position().column() as i32,
                 single_line_no_indent: false,
                 multi_line_indent_all: false,
             });
             // Mark all sub-nodes as having this node as a "leaf parent"
             self.mark_leaf_parent(node, node.id());
+            // Test the node for comments
+            // `comments` is sorted in reverse order, so `pop()` gets the first one.
+            while let Some(comment) = comments.pop() {
+                let node_section: InputSection = node.into();
+                // REVIEW: the double borrow is sort of weird, is this idiomatic?
+                if node_section.contains(&(&comment).into()) {
+                    match comment.commented {
+                        Commented::CommentedAfter { .. } => self
+                            .comments_before
+                            .entry(id)
+                            .or_default()
+                            .push_back((&comment).into()),
+                        Commented::CommentedBefore(_) => self
+                            .comments_after
+                            .entry(id)
+                            .or_default()
+                            .push_back((&comment).into()),
+                    }
+                } else {
+                    comments.push(comment);
+                    break;
+                }
```

Comment:
> All comments before/after, or just those immediately before/after?

I believe this code answers my question: It's the just comments immediately before/after...I think.